src/main/scala/com/databricks/labs/overwatch/env/Database.scala

```diff
@@ -27,18 +27,54 @@ class Database(config: Config) extends SparkSessionWrapper {
   }
 
   /**
-   * register an Overwatch target table in the configured Overwatch deployment
+   * Checks if the table has optimizeWrite enabled.
+   * @param target
+   * @return
+   */
+  private def checkAutoOptimizeWrite(target: PipelineTable): Boolean = {
+    import spark.implicits._
+    val optimiseWrite: Boolean = spark.sql(s"""describe detail '${target.tableLocation}'""")
+      .select("properties").as[Map[String, String]].collect()
+      .headOption.getOrElse(throw new Exception("string"))
+      .getOrElse("delta.autoOptimize.optimizeWrite", "false")
+      .toBoolean
+    optimiseWrite
+  }
```

Comment:
seems like this function should not return whether or not the property is true or false but rather whether or not it matches the target config, right?

shouldn't it return
`optimizeWrite == target.autoOptimize`?

I think you're trying to get too fancy with what this function does. Perhaps two functions would be easier to follow. 

1. Is the target actually optimizeWrite
2. Does the target actual setting match it's config